benchadapt/python/benchadapt/result.py

```diff
@@ -167,6 +172,16 @@ def to_publishable_dict(self):
                 "Result not publishable! `stats` and/or `error` must be be specified"
             )
 
+        if not (
+            res_dict["github"].get("repository") and res_dict["github"].get("commit")
+        ):
+            raise ValueError(
```

Comment:
This one on BenchmarkResult _might_ be a little too strict, for the workflow where someone first creates a Run manually and then wants to post results to it without worrying about the `github` object (since `github` is ignored on results if the `run_id` already exists). But I'm okay leaving this as-is and handling that issue later if it pops up.

Reply:
Yeah, that's true. We're already sort of moving in this direction—before we could post runs directly, all the run metadata had to be on the result. Now that we've normalized a bit, it would make sense to just take `run_id` in cases where the run is sent first, but we've still got a ton of legacy code that isn't opening runs first that still needs the duplicated fields. But as all that gradually gets migrated, we should at least think about deprecating `run_name`, `run_reason`, `github`, etc.

For now, I don't really mind this particular duplication, because if users are using env vars to populate this on the run, it will get picked up here without them doing anything, and benchconnect checks that they don't disagree (I believe missing is fine there). But we could reduce this one to a warning and keep the run one an error while we're here anyway.

Reply:
changed